extension/app/ts/background/metadataUtils.ts

```diff
@@ -9,38 +9,66 @@ import { GetAddressBookDataFilter } from '../utils/interceptor-messages.js'
 const LOGO_URI_PREFIX = `../vendor/@darkflorist/address-metadata`
 
 export function getMetadataForAddressBookData(filter: GetAddressBookDataFilter, addressInfos: readonly AddressInfo[] | undefined) {
+	const searchString = filter.searchString !== undefined && filter.searchString.trim().length > 0 ? filter.searchString.trim().toLowerCase() : undefined
 	switch(filter.filter) {
 		case 'My Contacts': return { data: [], length: 0 }
-		case 'My Active Addresses': return {
-			data: addressInfos ? addressInfos.slice(filter.startIndex, filter.maxIndex).map( (x) => ({
-				type: 'addressInfo' as const,
-				...x
-			})) : [],
-			length: addressInfos ? addressInfos.length : 0
+		case 'My Active Addresses': {
+			const searchFiltered = searchString === undefined ? addressInfos : addressInfos?.filter( (x) =>
+				x.address.toString().includes(searchString)
+				|| x.name.toLowerCase().includes(searchString)
+			)
```

Comment:
Consider fuzzy matching instead.  Fuzzy matching provides a better filter experience for users and is relatively easy to implement.

https://blog.amjith.com/fuzzyfinder-in-10-lines-of-python